### PR TITLE
Extract scalastyle workflow to a single yml and disable it in other jobs

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

quick fail on style issue and reduce ga overhead

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

```
=========dev/kyuubi-tpcds/target/scalastyle-output.xml=========
=========dev/kyuubi-extension-spark-3-1/target/scalastyle-output.xml=========
=========dev/kyuubi-extension-spark-common/target/scalastyle-output.xml=========
=========dev/kyuubi-extension-spark-3-2/target/scalastyle-output.xml=========
=========externals/kyuubi-download/target/scalastyle-output.xml=========
=========externals/kyuubi-trino-engine/target/scalastyle-output.xml=========
=========externals/kyuubi-spark-sql-engine/target/scalastyle-output.xml=========
=========externals/kyuubi-flink-sql-engine/target/scalastyle-output.xml=========
=========kubernetes/integration-tests/target/scalastyle-output.xml=========
=========kyuubi-assembly/target/scalastyle-output.xml=========
=========kyuubi-common/target/scalastyle-output.xml=========
  <error line="22" source="org.scalastyle.file.FileLineLengthChecker" severity="error" message="File line length exceeds 100 characters"/>
=========kyuubi-ctl/target/scalastyle-output.xml=========
=========kyuubi-ha/target/scalastyle-output.xml=========
=========kyuubi-metrics/target/scalastyle-output.xml=========
=========kyuubi-server/target/scalastyle-output.xml=========
=========kyuubi-zookeeper/target/scalastyle-output.xml=========
=========target/scalastyle-output.xml=========
=========tools/spark-block-cleaner/target/scalastyle-output.xml=========

```

### CODE REVIEW

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup JDK 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: zulu
+          java-version: 8
+      - name: Scalastyle with Maven
+        run: build/mvn scalastyle:check -Ptpcds -Pspark-block-cleaner -Pkubernetes -Pspark-3.1 -Pspark-3.2
```
Comment (pan3793) [inline .github/workflows/style.yml:41]: `-Pspark-3.1 -Pspark-3.2` should be conflict with each other on compiling, but it should be ok for scalastyle check

**Review by pan3793 — APPROVED** [review]

File: .github/workflows/style.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup JDK 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: zulu
+          java-version: 8
+      - name: Scalastyle with Maven
+        run: build/mvn scalastyle:check -Ptpcds -Pspark-block-cleaner -Pkubernete -Pspark-3.1 -Pspark-3.2
```
Comment (cfmcgrady) [inline .github/workflows/style.yml:41]: `-Pkubernete` -> `-Pkubernetes`

Comment (yaooqinn) [reply .github/workflows/style.yml:41]: 👍

**Comment by cfmcgrady** [comment]
BTW, do we have a Java-style check plugin?

**Comment by yaooqinn** [comment]
> BTW, do we have a Java-style check plugin?

yes, are you interested in adding on one more job or step for java after this

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**